app/src/main/java/com/appsubaruod/comicviewer/model/ComicModel.java

```diff
@@ -91,37 +95,32 @@ public void run() {
 
     private void copyToAppStorage(Uri uri) {
         initialize();
-        String path = mFileOperator.getPath(uri);
+        String path = mFileOrganizer.getPath(uri);
         if (path == null) {
             Log.d(LOG_TAG, "Unsupported uri. Maybe network storage: " + uri.toString());
             Log.d(LOG_TAG, "try to open");
             try {
-                ContentResolver cR = mContext.getContentResolver();
-                MimeTypeMap mime = MimeTypeMap.getSingleton();
-                String extensionFromMimeType = mime.getExtensionFromMimeType(cR.getType(uri));
-                Log.d(LOG_TAG, "mime-type : " + extensionFromMimeType);
-                if ("zip".equals(extensionFromMimeType)) {
-                    mFileOperator.unpackZip(uri);
-                } else {
-                    Log.w(LOG_TAG, "this type of file is not supported now!");
-                }
+                mFileOrganizer.requestNetworkContent(uri);
             } catch (Exception e) {
                 e.printStackTrace();
             } finally {
                 return;
             }
         }
+        copyLocalFile(uri, path);
+    }
+
+    private void copyLocalFile(Uri uri, String path) {
```

Comment:
Change method name, since ComicModel just requests the actual content and does not have concern on how to be resolved on the FileOrganizer instance.